docs/src/contributing/testnet.md

```diff
@@ -0,0 +1,266 @@
+  * On other cloud platforms the testnet will not self-destruct!
+* To enable metrics in the testnet, at this point you need to either:
+  * `./init-metrics.sh -c testnet-dev-${user} ${user}` to create a new metrics database from scratch
+  * Manually set `SOLANA_METRICS_CONFIG` in `./net/config/config` (which is exactly what `init-metrics.sh` does for you)
+  * `./init-metrics.sh -e` to load metrics config from `SOLANA_METRICS_CONFIG` into the testnet config file or
+* `./net.sh` controls the payload on the testnet nodes, i.e. bootstrapping, the validators and bench-tps. In principle, you can run everything by hand, but `./net.sh` makes it easier.
+  * `./net.sh start` to actually run the test network.
+    * This will actually upload your current sources to the bootstrap host, build them there and upload the result to all the nodes
+    * The script will take 5-10 of minutes to run, in the end it should print something like
+     ```
+     --- Deployment Successful
+     Bootstrap validator deployment took 164 seconds
+     ```
+    * You can also make sure it logs successful test transfers:
+    ```✅ 1 lamport(s) transferred: seq=0   time= 402ms signature=33uJtPJM6ekBGrWCgWHKw1TTQJVrLxYMe3sp2PUmSRVb21LyXn3nDbQmzsgQyihE7VP2zD2iR66Du8aDUnSSd6pb```
+  * `./net.sh start  bench-tps=2="--tx_count 2500"` will start 2 clients with bench-tps workload sending 2500 transactions per batch.
+    * --tx_count argument is passed to the bench-tps program, see its manual for more options
+  * `./net.sh sanity`  to test the deployment, it is also run by start command
+  * `./net.sh stop`  to stop the validators and client. This does not kill the machines, so you can study the logs etc.
+  * `./net.sh start --nobuild` will skip the source compilation, you will generally want that if you are only changing configuration files rather than code, or just want to re-run the last test.
+* To connect to the nodes:
+  * `./gce.sh info ` to get the public IPs
+  * `./ssh.sh <IP> ` to get a shell on the node
+  * `sudo su` will give you root access on the nodes
+  * Nodes run latest ubuntu LTS image
+* You can also interact with the nodes using solana cli:
+```bash
+# source ip list  use as ${validatorIpList[4]}
+source net/config/config
+
+# airdrop
+../target/release/solana -u http://${validatorIpList[1]}:8899 airdrop 1
+
+# check feature
+../target/release/solana -u http://${validatorIpList[1]}:8899 feature status
+
+# activate a feature
+../target/release/solana -u http://${validatorIpList[1]}:8899 feature activate <path to .json>
+
+# check the stakes on current validators
+../target/release/solana --url http://${validatorIpList[0]}:8899 validators
+```
+
+## Tips
+
+### Automation
+You will want to have a script like this pretty much immediately to avoid making mistakes in the init process:
+```bash
+# Create the testnet with reasonable node sizes for a small test
```

Comment:
I think comment is needed for this PR: this command creates cluster with 7 nodes, 5 are validators and 2 are for clients.

Reply:
done